am5-compile/compile.yaml

```diff
@@ -93,13 +94,29 @@ compile:
                   "mom6/ocean_BGC/mocsy/src", 
                   "mom6/MOM6-examples/src/MOM6/pkg/GSW-Fortran/*/", 
                   "mom6/MOM6-examples/src/MOM6/config_src/infra/FMS2"] 
-          cppdefs: "-DINTERNAL_FILE_NML" #-DMAX_FIELDS_=100 -DNOT_SET_AFFINITY -D_USE_MOM6_DIAG -D_USE_GENERIC_TRACER  -DUSE_PRECISION=2"
+          cppdefs: "-DINTERNAL_FILE_NML -DMAX_FIELDS_=100 -DNOT_SET_AFFINITY -D_USE_MOM6_DIAG -D_USE_GENERIC_TRACER  -DUSE_PRECISION=2"
           makeOverrides: 'OPENMP=""'
           otherFlags: !join [*FMSincludes, " ", *momIncludes]
+          additionalInstructions: |
+              cd mom6
+              git clone https://github.com/NOAA-GFDL/MOM6-examples.git -b dev/gfdl
+              cd MOM6-examples
+              git checkout 753f135 # this is just the value of MOM6_GIT_TAG property, can't seem to use variable
+              git submodule update --recursive --init src/MOM6 src/SIS2 src/icebergs
+
+              cd src/MOM6
+              git checkout 878fd1ef671456a804c598df606e4cf7c803c203
+
+              cd ../SIS2
+              git checkout 00c3b5be13721a7af99f62f2163969e8dd303df7
+
+              cd ../icebergs
+              git checkout 6a8944fc7544608942de9a42bd3b96f6e56684e0
+              cd ../../../..
```

Comment:
how about setting the `MOM6_GIT_TAG` property in the am5.yaml, as

```
fre_properties:
    &MOM6_GIT_TAG        "foo"
```

and then using it here as

```
              cd src/MOM6
              git checkout *MOM6_GIT_TAG
```

Reply:
This does not seem to work:
This is checkout script that is generated: /ncrc/home1/Uriel.Ramirez/fremake_canopy/test/am5f8d0r1_compile/src/checkout.sh
```
cd MOM6-examples
git checkout *MOM6_GIT_TAG #753f135 this is just the value of MOM6_GIT_TAG property, can't seem to use variable
git submodule update --recursive --init src/MOM6 src/SIS2 src/icebergs

```

Reply:
It looks like all of the `additional_instructions` are read in as a string with the use of `|`. If you did want to specify tags in the `fre_properties`, it looks like this format might work using the `!join` constructor:
```
          additionalInstructions: !join ["cd mom6\n",
                                         "git clone https://github.com/NOAA-GFDL/MOM6-examples.git -b dev/gfdl\n",
                                         "cd MOM6-examples\n",
                                         "git checkout 753f135\n",
                                         "git submodule update --recursive --init src/MOM6 src/SIS2 src/icebergs\n",
                                         "cd src/MOM6\n",
                                         "git checkout ",*MOM6_GIT_TAG,
                                         "cd ../SIS2\n",
                                         "git checkout 00c3b5be13721a7af99f62f2163969e8dd303df7\n",
                                         "cd ../icebergs\n",
                                         "git checkout 6a8944fc7544608942de9a42bd3b96f6e56684e0\n",
                                         "\ncd ../../../.."]

```

Reply:
that works https://github.com/NOAA-GFDL/fre-examples/pull/1/commits/199169c3e158f561e11636fea6ec0915251fb029